packages/frontend-2/components/workspace/ProjectList.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div>
+    <div v-if="!showEmptyState" class="flex flex-col gap-4">
+      <WorkspaceHeader
+        v-if="workspace"
+        :icon="Squares2X2Icon"
+        :workspace-info="workspace"
+        :project-count="projects?.totalCount"
+      />
+      <div class="flex flex-col sm:flex-row gap-2 sm:items-center justify-between">
+        <div class="flex flex-col sm:flex-row gap-2">
+          <FormTextInput
+            name="modelsearch"
+            :show-label="false"
+            placeholder="Search..."
+            :custom-icon="MagnifyingGlassIcon"
+            color="foundation"
+            wrapper-classes="grow md:grow-0 md:w-60"
+            :show-clear="!!search"
+            :model-value="bind.modelValue"
+            v-on="on"
+          ></FormTextInput>
+          <!-- <FormSelectProjectRoles
+            v-if="!showEmptyState"
+            v-model="selectedRoles"
+            class="md:w-56 grow md:grow-0"
+            fixed-height
+          /> -->
+        </div>
+        <FormButton v-if="!isGuest" @click="openNewProject = true">
+          New project
+        </FormButton>
+      </div>
+    </div>
+
+    <CommonLoadingBar :loading="showLoadingBar" class="my-2" />
+
+    <ProjectsDashboardEmptyState
+      v-if="showEmptyState"
+      @create-project="openNewProject = true"
+    />
+
+    <template v-else-if="projects?.items?.length">
+      <ProjectsDashboardFilled :projects="projects" />
+      <InfiniteLoading
+        :settings="{ identifier: infiniteLoaderId }"
+        @infinite="infiniteLoad"
+      />
+    </template>
+
+    <CommonEmptySearchState v-else-if="!showLoadingBar" @clear-search="clearSearch" />
+
+    <ProjectsAddDialog v-model:open="openNewProject" :workspace-id="workspaceId" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { MagnifyingGlassIcon, Squares2X2Icon } from '@heroicons/vue/24/outline'
+import { useQuery, useQueryLoading } from '@vue/apollo-composable'
+import { useActiveUser } from '~~/lib/auth/composables/activeUser'
+import type { InfiniteLoaderState } from '~~/lib/global/helpers/components'
+import type { Nullable, Optional, StreamRoles } from '@speckle/shared'
+import { workspacePageQuery } from '~~/lib/workspaces/graphql/queries'
+import { useDebouncedTextInput } from '@speckle/ui-components'
+
+const logger = useLogger()
+
+const infiniteLoaderId = ref('')
+const cursor = ref(null as Nullable<string>)
+const selectedRoles = ref(undefined as Optional<StreamRoles[]>)
+const openNewProject = ref(false)
+const showLoadingBar = ref(false)
+
+const { isGuest } = useActiveUser()
+const areQueriesLoading = useQueryLoading()
+
+const props = defineProps<{
+  workspaceId: string
+}>()
+
+const {
+  on,
+  bind,
+  value: search
+} = useDebouncedTextInput({
+  debouncedBy: 800,
+  model: ref('')
+})
+
+const {
```

Comment:
Simplify all of this pagination stuff w/ `usePaginatedQuery`. You can leave ProjectsDashboard.vue as is since its an old page, but this is a new one.